components/contribution-flow/StepDetails.js

```diff
@@ -213,6 +226,33 @@ const StepDetails = ({
           )}
         </StyledInputField>
       )}
+      <Modal show={showModal} onClose={() => setShowModal(false)} height="300px">
+        <ModalHeader>
+          <FormattedMessage id="Frequency.change" defaultMessage={'Change frequency?'} />
+        </ModalHeader>
+        <ModalBody>
+          <FormattedMessage
+            id="contribute.changeFrequency.confirmMsg"
+            defaultMessage={
+              "If you're changing the frequency, we will not contribute to this specific tier 'Gold Sponsor'"
+            }
+          />
+        </ModalBody>
+        <ModalFooter>
+          <StyledButton mx={20} onClick={() => setShowModal(false)}>
+            <FormattedMessage id="collective.archive.cancel.btn" defaultMessage={'Cancel'} />
+          </StyledButton>
+          <StyledButton
+            buttonStyle="primary"
+            onClick={() => {
+              setShowModal(false);
+              dispatchChange({ interval: tempInterval });
```

Comment:
What was asked was not to just change the state, but also to reset the tier and land in the "custom financial contribution". How to check that it works?

- We should not see `"Contribute to "{tierName}" tier"` anymore in the screen
- The URL should not be `/{collectiveSlug}/contribute/{tierSlug}/checkout/details` anymore but `/{collectiveSlug}/donate/details`